vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py

```diff
@@ -0,0 +1,331 @@
+
+    def test_create_range_slider_mandatory_only(self):
+        range_slider = vm.RangeSlider()
+
+        assert hasattr(range_slider, "id")
+        assert range_slider.type == "range_slider"
+        assert range_slider.min is None
+        assert range_slider.max is None
+        assert range_slider.step is None
+        assert range_slider.marks is None
+        assert range_slider.value is None
+        assert range_slider.title is None
+        assert range_slider.actions == []
+
+    def test_create_range_slider_mandatory_and_optional(self):
+        range_slider = vm.RangeSlider(min=0, max=10, step=1, marks={}, value=[1, 9], title="Test title")
+
+        assert range_slider.min == 0
+        assert range_slider.max == 10
+        assert range_slider.step == 1
+        assert range_slider.value == [1, 9]
+        assert range_slider.title == "Test title"
+        assert range_slider.actions == []
+
+    @pytest.mark.parametrize(
+        "min, max",
+        [
+            (0, None),
+            (None, 10),
+            (0, 10),
+        ],
+    )
+    def test_valid_min_max(self, min, max):
+        range_slider = vm.RangeSlider(min=min, max=max)
+        min = int(min) if isinstance(min, str) else min
+        max = int(max) if isinstance(max, str) else max
+
+        assert range_slider.min == min
+        assert range_slider.max == max
+
+    def test_invalid_min_max(self):
+        with pytest.raises(
+            ValidationError, match="Maximum value of slider is required to be larger than minimum value."
+        ):
+            vm.RangeSlider(min=10, max=0)
+
+    @pytest.mark.parametrize("value", [None, [1, 2], [0.1, 1.1], [-10, 10], [10, 10]])
+    def test_valid_value(self, value):
+        range_slider = vm.RangeSlider(min=-10, max=10, value=value)
+
+        assert range_slider.value == value
+
+    @pytest.mark.parametrize(
+        "value, match",
+        [
+            ([0], "ensure this value has at least 2 items"),
+            ([], "ensure this value has at least 2 items"),
+            (2, "value is not a valid list"),
+            ([0, None], "1 validation error for RangeSlider"),
+            ([None, None], "2 validation errors for RangeSlider"),
+            ([-1, 11], "Please provide a valid value between the min and max value."),
+            ([1, 2, 3], "ensure this value has at most 2 items"),
+        ],
+    )
+    def test_invalid_value(self, value, match):
+        with pytest.raises(ValidationError, match=match):
+            vm.RangeSlider(min=0, max=10, value=value)
+
+    @pytest.mark.parametrize(
+        "step",
+        [1, 2.5, 10, None],
+    )
+    def test_valid_step(self, step):
+        range_slider = vm.RangeSlider(min=0, max=10, step=step)
+
+        assert range_slider.step == step
+
+    def test_invalid_step(self):
+        with pytest.raises(ValidationError, match="The step value of the slider must be less than the max value."):
+            vm.RangeSlider(min=0, max=10, step=11)
```

Comment:
In relation to the other comment: what is `min` was -4, this is valid no?

Reply:
Yes, I've made changes to the validator to account for that.